File: web/src/lib/email.ts
import { ALLOWED_EMAIL_DOMAIN } from "./constants";

export function normalizeEmail(email: string) {
  return email.trim().toLowerCase();
}

export function isInstituteEmail(email: string) {
  const normalized = normalizeEmail(email);
  const [local, domain] = normalized.split("@");
  return Boolean(local) && domain === ALLOWED_EMAIL_DOMAIN;
}

export function adminEmails() {
  return (process.env.ADMIN_EMAILS ?? "shubham.pgpex26@iimshillong.ac.in")
    .split(",")
    .map((value) => value.trim().toLowerCase())
    .filter(Boolean);
}

export function isAdminEmail(email: string) {
  return adminEmails().includes(normalizeEmail(email));
}

File: lib/utils/email.dart
import 'constants.dart';

bool isInstituteEmail(String email) {
  final normalized = email.trim().toLowerCase();
  return normalized.endsWith('@${AppConstants.allowedEmailDomain}') &&
      normalized.split('@').first.isNotEmpty;
}

String normalizeEmail(String email) => email.trim().toLowerCase();
